MVC-App/Controllers/RelationsController.cs

```diff
@@ -8,163 +8,33 @@
-                            if (char.IsLetter(value[valueIterator]))
-                            {
-                                result.Add(char.ToUpper(value[valueIterator]));
-
-                                valueIterator++;
-                            }
-                            else return value;
-
-                            break;
-                        }
-                    case 'l':
-                        {
-                            if (char.IsLetter(value[valueIterator]))
-                            {
-                                result.Add(char.ToLower(value[valueIterator]));
-
-                                valueIterator++;
-                            }
-                            else return value;
-
-                            break;
-                        }
-                    default:
-                        {
-                            result.Add(mask[i]);
-                            if (!char.IsLetterOrDigit(value[valueIterator]))
-                            {
-                                if (value[valueIterator] == mask[i])
-                                {
-                                    valueIterator++;
-                                }
-                                else return value;
-                            }
-                            break;
-                        }
-                }
-            }
-            if (value.Length == valueIterator)
-            {
-                return new string(result.ToArray());
-            }
-            else return value;
+            IKernel ninjectKernel = new StandardKernel();
+            ninjectKernel.Bind<IRelationService>().To<RelationService>();
+            relationService = ninjectKernel.Get<IRelationService>();
         }
+        
 
         // GET: tblRelations
         public async Task<ActionResult> Index(Guid? categoryId)
         {
-            var countries = db.tblCountry.ToList();
-
-            var relationModels = await InitRelationModels();
-
-            if(categoryId != null && categoryId != Guid.Empty)
-            {
-                relationModels = relationModels.Where(p => p.Categories.Contains(categoryId.Value)).ToList();
-            }
-
-            var relationListVM = new RelationListViewModel { RelationViewModels = relationModels, Categories = categoryFilter, Countries = countries };
-
-            return View(relationListVM);
-        }
-
-        // GET: tblRelations/Details/5
-        public async Task<ActionResult> Details(Guid? id)
-        {
-            if (id == null)
-            {
-                return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
-            }
-
-            tblRelation tblRelation = await db.tblRelation.FindAsync(id);
-
-            if (tblRelation == null)
-            {
-                return HttpNotFound();
-            }
-
-            return View(tblRelation);
+            return View(await relationService.GetAsync(categoryId));
```

Comment:
I've already told you. It is a bad practise to put everything in one line. 
Separate it.